src/core/test/system/borrower/balance.t.sol

```diff
@@ -0,0 +1,132 @@
+    }
+
+    function assertPostConditionTake(uint takeAmount, uint initialShelfBalance) public {
+        // assert: takeAmount > 0
+        assert(takeAmount > 0);
+        // assert: all funds transferred from tranche reserve
+        assertEq(currency.balanceOf(address(junior)), 0);
+        // assert: shelf received funds
+        assertEq(currency.balanceOf(address(shelf)), add(initialShelfBalance, takeAmount));
+    }
+
+    function assertPreConditionGive(uint giveAmount) public {
+        // assert: borrowFromTranches is inactive
+        assert(!distributor.borrowFromTranches());
+        // assert: shelf has funds
+        assert(currency.balanceOf(address(shelf)) == giveAmount);
+    }
+
+    function assertPostConditionGive(uint giveAmount, uint initialJuniorBalance) public {
+        // assert: giveAmount > 0
+        assert(giveAmount > 0);
+        // assert: all funds transferred from shelf
+        assertEq(currency.balanceOf(address(shelf)), 0);
+        // assert: junior received funds
+        assertEq(currency.balanceOf(address(junior)), add(initialJuniorBalance, giveAmount));
+    }
+
+    function testBalanceTake() public {
+        uint takeAmount = 100 ether;
+        // supply junior tranche with funds
+        supplyFunds(takeAmount, address(junior));
+        assertPreConditionTake(takeAmount);
+        // take money from tranche and transfer into shelf
+        balanceTake();
+    }
+
+    function testFailBalanceTakeNoFundsAvailable() public {
+        // junior tranche not supplied with capital
+        balanceTake();
+    }
+
+    function testFailBalanceTakeInactive() public {
+        uint takeAmount = 100 ether;
+        // supply junior tranche with funds
+        supplyFunds(takeAmount, address(junior));
+        // deactivate borrow
+        distributor.file("borrowFromTranches", false);
+        balanceTake();
+    }
+
+    function testBalanceGive() public {
+        uint giveAmount = 100 ether;
+        // supply shelf with funds
+        supplyFunds(giveAmount, address(shelf));
+        // deactivate borrow
+        distributor.file("borrowFromTranches", false);
+        // take money from shelf and transfer into tranche
+        balanceGive();
+    }
+
+    function testFailBalanceGiveShelfHasNoFunds() public {
+        distributor.file("borrowFromTranches", false);
+        // do not supply shelf with funds
+        balanceGive();
+    }
+
+    function testFailBalanceGiveInactive() public {
+         uint giveAmount = 100 ether;
+        // supply shelf with funds
+        supplyFunds(giveAmount, address(shelf));
+        // do not deactivate borrow
+        balanceGive();
+    }
+
+    // Helper to supply shelf or tranches with currency without using supply or repay, since these functions are usign balance internally.
+    function supplyFunds(uint amount, address addr) public {
+        currency.mint(address(this), amount);
+        currency.transferFrom(address(this), address(addr), amount);
```

Comment:
I think this can be done in one line.

` currency.mint(address(addr), amount);`